### PR TITLE
Fix incorrect format for graphql operations

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
## Description

Fix Prettier incorrect format comments for graphql operations #12358

Before Prettier used to move comments to a new line...
```graphql
query {
  # some comment
  foo
}
```
When Prettier should have formatted them like so...
```graphql
query { # some comment
  foo
}
```

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by Harry-Hopkinson** [comment]
@fisker I now slice the input and check for ```query{#```

**Comment by Harry-Hopkinson** [comment]
It failed due to an API key.

**Comment by fisker** [comment]
I tried my best to not hurt your feelings, but you are wasting your and my time.

**Prettier pr-12703**
[Playground link](https://deploy-preview-12703--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBHArnATgTwATB4DEeAzhALZx6QVWwA6UeeAZhBADQBGAhlk05MWLAL5MmGbDibBhxMpWq16Meey58BUHrwBe8kSNEhOICAAcYAS2ilkoflggB3AAr8E9lLwA2L3hx7M24sXjAAazgYAGVeKgAZayg4ZFY-UjgQsMjomItw5IBzZBgsTDM4Cm44ABNauoTeKCL0XiK4ADEILApeGBsW5BBedBgIUxAACxgKXwB1Ket4UgKwOBivZesAN2WcYbBSYJBkzKwYNzCivrSMrJAAK1IADxji3zgARXQIeDvfJkzAUsOdhkUwhYpqhfJMLFhkjB5tZajApsgABwABmBzky80hw3hcHOO1SZgwfzgV0s3hGpAAtCk6nVJlg4BhrOyru1bkh0oCHpkKNZSuUhR9vr9-vz7mYYLxuMjUeikAAmeVhay+YoAYUofJAJIArJN0JkACqK7wCoEgHaYACSUAasBiYARVgAgi6YjAcJ8AZlRKIgA)

```sh
--parser graphql
```

**Input:**

```graphql
query { # some comment
  foo,bar
,
    }

query
{
  # some comment
  foo,bar
,baz
        }
```

**Output:**

```graphql
query { # some comment
  foo,bar
,
    }

query
{
  # some comment
  foo,bar
,baz
        }

```

File: src/language-graphql/printer-graphql.js [hunk 1]
```diff
@@ -7,6 +7,18 @@ const { isNextLineEmpty, isNonEmptyArray } = require("../common/util.js");
 const { insertPragma } = require("./pragma.js");
 const { locStart, locEnd } = require("./loc.js");
 
+function hasQueryComment(options, node) {
+  if (
+    options.originalText
+      .slice(locStart(node), locEnd(node))
+      .replace(/\s/g, "")
+      .includes("query{#")
```
Comment (B2o5T) [inline src/language-graphql/printer-graphql.js:15]: Nice @Harry-Hopkinson, thanks for yours attempts and courage for fixing actual problem, but imagine if you have named query? Or unnamed/named subscription or mutation or query without `query` keyword, I think you can start by adding failling tests to found a way to fix it 👍

**Comment by thorn0** [comment]
Closing this because https://github.com/prettier/prettier/issues/12358 has been closed.

Thanks for trying to help. Please note that PRs are accepted only for issues that have been triaged and labeled as [`type:bug`](https://github.com/prettier/prettier/issues?q=is%3Aissue+is%3Aopen+label%3Atype%3Abug), [`type:enhancement`](https://github.com/prettier/prettier/issues?q=is%3Aissue+is%3Aopen+label%3Atype%3Aenhancement), etc. If an issue is labeled as `status:needs discussion`, PRs aren't accepted.

**Pull request closed without merging**